### DISCUSSION TITLE
Seq with Distance

### DISCUSSION THREAD

**Discussion by mdorkhah** [GateNLP/python-gatenlp#112] [Help Q&A]
Is it possible to limit the maximum distance between the tokens in Seq?
Something like:
Seq(Ann(type="type1", name="tok1"), Ann(type="type1", name="tok2"), distance=100)

**Comment by johann-petrak** [comment]
What do you mean by "distance"?

Wen matching a Seq of two anns as in your example, then those annotations must follow immediately one after the other in the sequence of annotations generated from the input set. 

One way to restrict this is by including e.g. Tokens in the input set and allowing a certain number of tokens between the annotations of interest, e.g. 

`Seq(Ann(type="type1"), N(Ann(type="Token", min=0, max=3), Ann(type="type2"))` 

If you mean distance as in difference in offsets, this is not directly supported at the moment but may be an interesting feature to implement.
